src/lua/init.lua

```diff
@@ -206,8 +206,39 @@ rawset(package, "setsearchroot", setsearchroot)
 local compat = require('compat')
 package.loaded.compat = nil
 
+-- Execute scripts or load modules pointed by TT_PRELOAD
+-- environment variable.
+local function run_preload()
+    local tt_preload = os.getenv('TT_PRELOAD') or ''
+    assert(type(tt_preload) == 'string')
+    if #tt_preload == 0 then
+        return
+    end
+    for _, script in ipairs(tt_preload:split(';')) do
+        -- luacheck: ignore 542 empty if branch
+        if #script == 0 then
+            -- Ignore empty entries to allow duplicated semicolons
+            -- and leading/trailing semicolons.
+            --
+            -- It simplifies construction of the environment
+            -- variable value using concatenation.
+        elseif script:endswith('.lua') then
+            local fn, err = loadfile(script)
+            if fn == nil then
+                error(err)
+            end
+            fn(script)
+        else
+            require(script)
+        end
+    end
+end
+
 return {
     uptime = uptime;
     pid = pid;
-    compat = compat
+    compat = compat;
+    _internal = {
+        run_preload = run_preload,
```

Comment:
Minor: Why do we need to store this function? I guess we need to fire this function once and remove it completely when it finishes.

Feel free to ignore.

Reply:
I don't like several things in such approach:

* We protect a user against *accidental* usage of an internal API. Here `internal` namespace is enough and `_internal` is even better, because it also removes the field from autocomplection.
* There is no sense to attempt to forbid reaching an internal API if a user (think that) (s)he really wants it. It is similar to how the class member visibility works in programming languages: it is possible to hack around, but hard to reach by a mistake.
* And, to be honest, if a user (think that) (s)he really wants to reach an internal API, it is better to make this access explicit (via `_internal`, not `debug`) and less fragile (same).
* I don't like the approach implemented for box fields: add them at early initialization, remove a bit later and then add back within `box.cfg()`. That's hell for one, who trying to decipher such code.
* At last, I don't like extra code, which doesn't add any value for a user (in my understanding of his/her needs).

This is my general approach for such things. Don't protect against intentional hacks, but protect from accidental usages of an internal API: i.e. don't mix private and public functions in one namespace (without a meaningful prefix at least).